src/net_processing.cpp

```diff
@@ -1451,8 +1452,10 @@ void static ProcessGetBlockData(CNode* pfrom, const Consensus::Params& consensus
                 assert(!"cannot load block from disk");
             pblock = pblockRead;
         }
-        if (inv.type == MSG_BLOCK)
-            connman->PushMessage(pfrom, msgMaker.Make(NetMsgType::BLOCK, *pblock));
+        if (pblock) {
+            if (inv.type == MSG_BLOCK)
+                connman->PushMessage(pfrom, msgMaker.Make(NetMsgType::BLOCK, *pblock));
+        }
```

Comment:
This is incorrect, the whole `if (inv.type == MSG_BLOCK) ... else if (inv.type == MSG_FILTERED_BLOCK) ... else if (inv.type == MSG_CMPCT_BLOCK) ...` thing should be wrapped into `if (pblock)` instead.

Reply:
bb6c6597651a97ea76fc73f9a6f0f68714e7b55f